lib/ABI/AddABIAwarenessToFunctionPrototypes.cpp

```diff
@@ -0,0 +1,210 @@
+/// \file FunctionTypeAnalysis.cpp
```

Comment:
* Wrong name
* You choose the most obscure names possible `:D`. Let's call these `ConvertToCABIFunctionType` and `ConvertToRawFunctionType` or `LiftToCABIFunctionType` and `LowerToRawFunctionType`, but *lift* is an already quite overloaded term.

Reply:
2. I try to choose the most verbose name possible. Sometimes it backfires. Yeah, you already suggested `lift` but I'm against it for the exact same reason. Hence I just named it *exactly* as what it does - infuses/removes ABI information from a function. Even if that's obscure.

In short, I'd hate for something as internal-feeling as `RawFunctionType` and `CABIFunctionType` to show up in analysis list!
It's as if we're leaking an implementation detail into the one of the most user-facing parts of our interface.

Reply:
It's not internal, it's part of the model which is our primary interface with the user.
"Oh so many RFTs! I don't like them, what can I do??? **list analyses** Wait! `ConvertToCABIFunctionType`!"

Reply:
IMO, in a perfect world, with the combination of DLA and clever ABI detection we would be able to make sure ~60% of our users never find out what an RFT is.

If only this was the perfect world.

But I guess your point is also valid. They probably going to see them way too often :sad_parrot:

Reply:
They are called `ConvertToCABIFunctionType` and `ConvertToRawFunctionType` now.